### PR TITLE
fix: file reader for windows

### PR DESCRIPTION
**Pull request by randhir-devtron**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/devtron-labs/devtron/issues/4751

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/hooks/FileReader.ts [hunk 1]
```diff
@@ -22,6 +23,12 @@ export const useFileReader = () => {
         if (!fileData || fileData.data == null || !validator) {
             return
         }
+
+        // If the MIME type is not there
+        // In case of windows, the mime type gives some unexpected results so it needs to be handle explicitly
+        if (!fileData.type) {
+            fileData.type = getFileMimeType(fileData)
```
Comment (AbhishekA1509) [inline src/components/common/hooks/FileReader.ts:30]: We should not and can not mutate state directly it set itself throught setState.

File: src/components/common/hooks/FileReader.ts [hunk 2]
```diff
@@ -22,6 +23,15 @@ export const useFileReader = () => {
         if (!fileData || fileData.data == null || !validator) {
             return
         }
+
+        // If the MIME type is not there
+        // In case of windows, the mime type gives some unexpected results so it needs to be handle explicitly
+        if (!fileData.type) {
+            setFileData((prevFileData) => ({
+                ...prevFileData,
+                type: getFileMimeType(fileData),
```
Comment (AbhishekA1509) [inline src/components/common/hooks/FileReader.ts:32]: Move it to readFile

File: src/components/scopedVariables/utils.tsx [hunk 3]
```diff
@@ -11,16 +11,16 @@ import {
     YAML_PARSE_ERROR_STATUS,
 } from './constants'
 
-export const getFileMimeType = (fileData: FileDataType): MIME_TYPE => {
-    const fileNameEndsWith = fileData.name
+export const getFileMimeType = (fileDataName: string): MIME_TYPE => {
+    const fileNameEndsWith = fileDataName
```
Comment (AbhishekA1509) [inline src/components/scopedVariables/utils.tsx:15]: Remove this variable

File: src/components/scopedVariables/utils.tsx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 // @ts-nocheck
 import yaml from 'yaml'
 import { ScopedVariablesDataType } from './types'
-import { FileReaderStatus, ValidatorType } from '../common/hooks/types'
+import { FileReaderStatus, ValidatorType, FileDataType } from '../common/hooks/types'
```
Comment (AbhishekA1509) [inline src/components/scopedVariables/utils.tsx:4]: Remove unused import of FileDataType

**Review by AbhishekA1509 — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**